### PR TITLE
Fix CI skipped job

### PR DESCRIPTION
**Pull request by potsrevennil**
Seemed to found a way to fix https://github.com/pq-code-package/mlkem-c-aarch64/issues/155 by using matrix exclude

### CODE REVIEW

File: .github/workflows/ci_ec2_reusable.yml [hunk 1]
```diff
@@ -95,51 +95,63 @@ jobs:
...
+          - {tests: {
+              name: CBMC,
+              action: cbmc,
+              nix-shell: ci-cbmc,
+              run: false,
+            }}
     steps:
       - uses: actions/checkout@v4
-      - uses: ./.github/actions/cbmc
+      - uses: jenseng/dynamic-uses@v1
```
Comment (hanno-becker) [inline .github/workflows/ci_ec2_reusable.yml:139]: Why is this needed?

Comment (potsrevennil) [reply .github/workflows/ci_ec2_reusable.yml:139]: as I pointed out in #155, currently with the `if` statement, github will still show the skipped jobs of a workflow in UI.

While with the `matrix` `exclude`, we can hide the skipped jobs from UI, but since the jobs we are running (functest/lint/cbmc) use different action, we'd need to dynamic load them.

And currently github doesn't support expression evaluation in `uses`, so the only way would be to write an action.yml on the fly as a middleware to dynamic pointing to use the specify action, which is what `jenseng/dynamic-uses` have done.

Comment (hanno-becker) [reply .github/workflows/ci_ec2_reusable.yml:139]: Ok!

Comment (potsrevennil) [reply .github/workflows/ci_ec2_reusable.yml:139]: btw, I haven't yet confirmed, but I think that when using matrix, if the whole matrix is excluded, there'd still be a dummy job executing (where all the matrix fields are evaluated to empty string).

In this case, I think we'd at least run one of functest/lint/cbmc job, so it'd be fine, but it's not the case for ec2 related job (which we'd like to exclude the jobs if triggered from fork). 

Therefore I didn't apply this to ec2 related jobs

**Review by hanno-becker — APPROVED** [review]
It's sad that such hard-to-read trickery is needed, but well... thanks Han for looking into it!
